File: Java_Generated/Java/untitled/src/conversion/category_theory/Desktop.java
package conversion.category_theory;

import conversion.category_theory.interfaces.ICategoryArrow;
import conversion.category_theory.interfaces.ICategoryObject;
import conversion.category_theory.interfaces.IDesktop;

import java.util.*;

public class Desktop  implements IDesktop
{
    protected List<ICategoryObject> objects = new ArrayList<>();
    protected List<ICategoryArrow> arrows = new ArrayList<>();

    protected Map<String, ICategoryObject> objectMap = new TreeMap<>();
    protected Map<String, ICategoryArrow> arrowMap = new TreeMap<>();


    /**
     * @return
     */
    @Override
    public List<ICategoryObject> getCategoryObjects() {
        return objects;
    }

    /**
     * @return
     */
    @Override
    public List<ICategoryArrow> getCategoryArrows() {
        return arrows;
    }

    /**
     * @param obj
     */
    @Override
    public void addCategoryObject(ICategoryObject obj) {
objects.add(obj);
objectMap.put(obj.getCategoryObjectName(), obj);

    }

    /**
     * @param arrow
     */
    @Override
    public void addCategoryArrow(ICategoryArrow arrow) {
arrows.add(arrow);
arrowMap.put(arrow.getArrowName(), arrow);
    }

    /**
     * @param name
     * @return
     */
    @Override
    public ICategoryObject getCategoryObject(String name) {
        ICategoryObject o = objectMap.get(name);
        return  o;
    }

    /**
     * @param name
     * @return
     */
    @Override
    public ICategoryArrow getCategoryArrow(String name) {
       ICategoryArrow a = arrowMap.get(name);
       return a;
    }
}

File: src/TypeScript/reactprojectpure/src/Library/Desktop.ts
/* eslint-disable no-var */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-explicit-any */

import { OwnNotImplemented } from "./ErrorHandler/OwnNotImplemented";
import { Performer } from "./Performer";
import type { ICategoryArrow } from "./Interfaces/ICategoryArrow";
import type { ICategoryObject } from "./Interfaces/ICategoryObject";
import type { ICheck } from "./Interfaces/ICheck";
import type { IDesktop } from "./Interfaces/IDesktop";
import type { IObject } from "./Interfaces/IObject";
import type { IInitializeTask } from "./Interfaces/IInitializeTask"

export class Desktop implements IDesktop, IObject
{
    getClassName(): string {
        return this.typeName;
    }

    imlplementsType(type: string): boolean {
        return this.types.indexOf(type) >= 0;
    }
    protected typeName: string = "Desktop";

    protected types: string[] = ["IObject", "IDesktop", "IComponentCollection", "IObjectCollection", "Desktop"];

    protected categoryObjects: ICategoryObject[] = [];

    protected categoryArrows: ICategoryArrow[] = [];

    protected objects: IObject[] = [];

    protected name!: string;

    protected arrow!: ICategoryArrow;


    protected source!: ICategoryObject;


    protected target!: ICategoryObject;

    protected check !: ICheck;

    protected mapObjects: Map<string, ICategoryObject> = new Map()


    protected performer: Performer = new Performer();


     async initializeTaksAsync(cancel: AbortController): Promise<void> {

        var init = [];
        var ii = this.performer.getByInterface(this, "IInitializeTask");
        for (var i of ii) {
            var k = i as unknown as IInitializeTask;
            k.initializeTaskAsync(cancel);
            init.push(k);
        }
        await Promise.all(init);
    }

    public async loadAsync(cancel: AbortController): Promise<void> {
        await this.initializeTaksAsync(cancel);
        this.finish();
    }

    public finish(): void {

    }

    getObjectCollection(): IObject[] {
        return this.objects
    }


     addObject(obj: IObject): void {
        this.objects.push(obj);
    }
    getObjects(): IObject[] {
        return this.objects;
    }
    setCheck(check: ICheck): void {
        this.check = check
    }
    getCheck() {
        return this.check;
    }


    getCategoryObject(name: string): ICategoryObject {
        for (var o of this.categoryObjects) {
            var n = o.getCategoryObjectName();
            if (n == name) {
                return o;
            }
        }
        throw new OwnNotImplemented("DESKTOP");
    }

    getCategoryObjects(): ICategoryObject[] {
        return this.categoryObjects;
    }
    getCategoryArrows(): ICategoryArrow[] {
        return this.categoryArrows;
    }
    addCategoryObject(obj: ICategoryObject): void {
        this.categoryObjects.push(obj);
    }
    addCategoryArrow(arr: ICategoryArrow): void {
        this.categoryArrows.push(arr);
    }
    getName(): string {
        return this.name;
    }


}